Document neutron agent cleanup (#941)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/proc_adopting-networker-services-to-the-data-plane.adoc

[NOTE]
Alternatively, you can include the Networker node set in the `nodeSets` list before you deploy the main `OpenStackDataPlaneDeployment` CR. You cannot add new node sets to the `OpenStackDataPlaneDeployment` CR after deployment.

> . Cleanup any {networking_first_ref} agents that are no longer running.

Reviewer: ```suggestion
. Clean up any {networking_first_ref} agents that are no longer running.
```

+

> In some cases there would be agents from the old dataplane which has been replaced or retired. The function these agents provided may be provided by a new agent running in {rhos_long}, or the agents function may have been replaced by other components. For example, DHCP Agents may no longer be needed, since OVN DHCP in {rhos_long} can provide this function.

Reviewer: ```suggestion
[NOTE]
In some cases, agents from the old data plane that are replaced or retired remain in {rhos_acro}. The function these agents provided might be provided by a new agent that is running in {rhos_acro}, or the function might be replaced by other components. For example, DHCP agents might no longer be needed, since OVN DHCP in {rhos_acro} can provide this function.
```
@hjensas Please confirm the accuracy of my rewrite. FYI {rhos_acro} means "RHOSO".

Author: Suggestion looks good, it is in the latest commit.

Done.

> .. List the agents. The list of agents varies depending on the services you had enabled:

Reviewer: ```suggestion
.. List the agents. The list of agents varies depending on the services you had enabled on your previous data plane:
```
@hjensas I just wanted to clarify where these services were enabled. Let me know if this edit is accurate.

+
----
$ oc exec openstackclient -- openstack network agent list
+--------------------------------------+------------------------------+--------------------------+-------------------+-------+-------+----------------------------+
| ID                                   | Agent Type                   | Host                     | Availability Zone | Alive | State | Binary                     |
+--------------------------------------+------------------------------+--------------------------+-------------------+-------+-------+----------------------------+

> | e5075ee0-9dd9-4f0a-a42a-6bbdf1a6111c | OVN Controller Gateway agent | controller-0.localdomain |                   | :-)   | UP    | ovn-controller             |

Reviewer: Just checking: Is : - ) in the Alive column really supposed to be there?

Author: Yes, the output uses `:-)` in the `Alive` field if the agent has reported it's state. And `XXX` for agents that did not report state - there is a timeout so when the agent has not reported for the given interval it will switch from `:-)` to `XXX`.

| 856960f0-5530-46c7-a331-6eadcba362da | DHCP agent                   | controller-1.localdomain | nova              | XXX   | UP    | neutron-dhcp-agent         |
| 8bd22720-789f-45b8-8d7d-006dee862bf9 | DHCP agent                   | controller-2.localdomain | nova              | XXX   | UP    | neutron-dhcp-agent         |
| e584e00d-be4c-4e98-a11a-4ecd87d21be7 | DHCP agent                   | controller-0.localdomain | nova              | XXX   | UP    | neutron-dhcp-agent         |
+--------------------------------------+------------------------------+--------------------------+-------------------+-------+-------+----------------------------+
----

> .. For any agent in the list with `XXX` in the `Alive` field verify the Host and Agent Type, if the functions of this agent is no longer required and the agent has been permanently stopped on the {rhos_prev_long} host. Delete the agent by running:

Reviewer: ```suggestion
.. If any agent in the list shows `XXX` in the `Alive` field, verify the Host and Agent Type, if the functions of this agent is no longer required, and the agent has been permanently stopped on the {rhos_prev_long} host. Then, delete the agent by running:
```
@hjensas So this step means that the customer has to verify 3 things:
1. The Host and Agent Type, which they can get from the above table, correct?
2. If the functions of this agent is no longer required - How would one verify this?
3. The agent has been permanently stopped on the {rhos_prev_long} host. - How would one verify this?

Author: Suggestion in latest commit.

For the questions:
1. Yes the Host and Agent Type is in the output table.
2. There is a number of different Agent Types, and they may or may not have moved or replaced. It depends on the Agent in question. I think if we want to elaborate on that we should ask the Networking DFG for guidance.
3. Previous steps in the adoption procedure would have been ran to stop and disable services, to validate one would have to log in to the host and run `systemctl status <service-name>` commands.

[NOTE]
Alternatively, you can include the Networker node set in the `nodeSets` list before you deploy the main `OpenStackDataPlaneDeployment` CR. You cannot add new node sets to the `OpenStackDataPlaneDeployment` CR after deployment.

. Clean up any {networking_first_ref} agents that are no longer running.
+
[NOTE]
In some cases, agents from the old data plane that are replaced or retired remain in {rhos_acro}. The function these agents provided might be provided by a new agent that is running in {rhos_acro}, or the function might be replaced by other components. For example, DHCP agents might no longer be needed, since OVN DHCP in {rhos_acro} can provide this function.

> .. List the agents.

Reviewer: ```suggestion
.. List the agents:
```

| ID                                   | Agent Type                   | Host                     | Availability Zone | Alive | State | Binary                     |
+--------------------------------------+------------------------------+--------------------------+-------------------+-------+-------+----------------------------+
| e5075ee0-9dd9-4f0a-a42a-6bbdf1a6111c | OVN Controller Gateway agent | controller-0.localdomain |                   | :-)   | UP    | ovn-controller             |
| 856960f0-5530-46c7-a331-6eadcba362da | DHCP agent                   | controller-1.localdomain | nova              | XXX   | UP    | neutron-dhcp-agent         |
| 8bd22720-789f-45b8-8d7d-006dee862bf9 | DHCP agent                   | controller-2.localdomain | nova              | XXX   | UP    | neutron-dhcp-agent         |
| e584e00d-be4c-4e98-a11a-4ecd87d21be7 | DHCP agent                   | controller-0.localdomain | nova              | XXX   | UP    | neutron-dhcp-agent         |
+--------------------------------------+------------------------------+--------------------------+-------------------+-------+-------+----------------------------+
----

> .. If any agent in the list shows `XXX` in the `Alive` field, verify the Host and Agent Type, if the functions of this agent is no longer required, and the agent has been permanently stopped on the {rhos_prev_long} host. Then, delete the agent by running:

Reviewer: ```suggestion
.. If any agent in the list shows `XXX` in the `Alive` field, verify the Host and Agent Type, if the functions of this agent is no longer required, and the agent has been permanently stopped on the {rhos_prev_long} host. Then, delete the agent:
```
